### PR TITLE
Upgrade 2.x deps

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

Upgrade deps to remove `npm audit` warnings.

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: test/helpers/browser.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ var vm = require('vm');
 function createContext() {
   var context = {
     // required by browserify
-    XMLHttpRequest: function() { throw new Error('not implemented'); },
```
Comment (b-admike) [inline test/helpers/browser.js]: Is this change also associated with the upgrade?

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+package-lock=true
```
Comment (b-admike) [inline .npmrc]: Do we need to enable package-lock here? Is that something we want to do throughout our repos?

File: test/helpers/browser.js (hunk 1 above)
Comment (raymondfeng) [reply test/helpers/browser.js]: Yes. Otherwise, it throws an error - not implemented

File: .npmrc (hunk 2 above)
Comment (raymondfeng) [reply .npmrc]: I enabled it to run `npm audit`.

**Comment by raymondfeng** [comment]
FYI: I'm not sure why this PR shows 63 commits.

**Pull request closed without merging**